1-JavaScript/Week3/homework/ex7-mindPrivacy.js

```diff
@@ -26,8 +27,14 @@ const employeeRecords = [
   },
 ];
 
-function filterPrivateData(/* parameter(s) go here */) {
+function filterPrivateData(employeeDetails) {
   // TODO complete this function
+  const publicData = [];
+  for (const { name, occupation, email } of employeeDetails) {
+    const newData = { name, occupation, email };
+    publicData.push(newData);
+  }
+  return publicData;
 }
```

Comment:
An alternative syntax is:

```suggestion
function filterPrivateData(arrayEmployees) {
  return arrayEmployees.map((employee) => {
    const {name, occupation, email} = employee
    return {name, occupation, email}
  })
}
```

Notice that there is more separation of logic; there is the "map" which does just that, it maps, then there is the destructing line, and then there is the returning of the new object.

Please note that my version is not better, it's just different, I hope you find it interesting.

Reply:
Thank you a lot Jasper for your feedback;
this will help me a lot to improve my skills and code, and I always look forward for that.

Of course after studying the high order functions and array methods then using them is better than using loop to go through arrays